Index: app/src/main/java/com/example/tinnhn/Call/GroupListActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/example/tinnhn/Call/GroupListActivity.java	(date 1606290676274)
+++ app/src/main/java/com/example/tinnhn/Call/GroupListActivity.java	(date 1606290676274)
@@ -0,0 +1,99 @@
+package com.example.tinnhn.Call;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.os.Bundle;
+import android.view.View;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ListView;
+
+import com.example.tinnhn.HoiThoai;
+import com.example.tinnhn.R;
+import com.google.firebase.database.ChildEventListener;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GroupListActivity extends AppCompatActivity {
+    private DatabaseReference mFirebaseDatabase;
+    private FirebaseDatabase mFirebaseInstance;
+    private DatabaseReference mDatabase;
+    String TenGroup;
+    int IdGroup;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.group_list);
+
+        final EditText tengrp=findViewById(R.id.tengroup);
+        Button them=findViewById(R.id.themgrp);
+        ListView grplist=findViewById(R.id.grplv);
+
+        mDatabase = FirebaseDatabase.getInstance().getReference();
+        ArrayList<String> dsgrp = new ArrayList<>();
+
+        mDatabase.child("Group").addChildEventListener(new ChildEventListener() {
+            @Override
+            public void onChildAdded(@NonNull DataSnapshot snapshot, @Nullable String previousChildName) {
+                String nhom=snapshot.getValue().toString();
+                dsgrp.add(nhom);
+            }
+
+            @Override
+            public void onChildChanged(@NonNull DataSnapshot snapshot, @Nullable String previousChildName) {
+
+            }
+
+            @Override
+            public void onChildRemoved(@NonNull DataSnapshot snapshot) {
+
+            }
+
+            @Override
+            public void onChildMoved(@NonNull DataSnapshot snapshot, @Nullable String previousChildName) {
+
+            }
+
+            @Override
+            public void onCancelled(@NonNull DatabaseError error) {
+
+            }
+        });
+
+        ArrayAdapter adapter = new ArrayAdapter(this, android.R.layout.simple_list_item_1, dsgrp);
+        grplist.setAdapter(adapter);
+
+
+
+        them.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Group group =new Group(tengrp.getText().toString(),RandomString(9));
+                mDatabase.child("Group").push().setValue(group);
+            }
+        });
+
+
+
+    }
+    public int RandomString(int n) {
+//        String AlphaNumericString = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvxyz0123456789";
+        String AlphaNumericString = "0123456789";
+        StringBuilder sb = new StringBuilder(n);
+        for (int i = 0; i < n; i++) {
+            int index = (int) (AlphaNumericString.length() * Math.random());
+            sb.append(AlphaNumericString.charAt(index));
+        }
+        return Integer.parseInt(sb.toString());
+    }
+}
\ No newline at end of file
Index: app/src/main/java/com/example/tinnhn/Call/Group.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/example/tinnhn/Call/Group.java	(date 1606212155239)
+++ app/src/main/java/com/example/tinnhn/Call/Group.java	(date 1606212155239)
@@ -0,0 +1,14 @@
+package com.example.tinnhn.Call;
+
+public class Group {
+    public String TenGroup;
+    public int IdGroup;
+
+    public Group() {
+    }
+
+    public Group(String TenGroup, int IdGroup) {
+        this.TenGroup = TenGroup;
+        this.IdGroup = IdGroup;
+    }
+}
Index: app/src/main/res/layout/group_list.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/group_list.xml	(date 1606195840697)
+++ app/src/main/res/layout/group_list.xml	(date 1606195840697)
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout
+    xmlns:android="http://schemas.android.com/apk/res/android" android:layout_width="match_parent"
+    android:layout_height="match_parent">
+    <ListView
+        android:id="@+id/grplv"
+        android:layout_width="match_parent"
+        android:layout_height="500dp"/>
+    <EditText
+        android:id="@+id/tengroup"
+        android:hint="nhap ten group"
+        android:layout_below="@+id/grplv"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"/>
+    <Button
+        android:id="@+id/themgrp"
+        android:layout_below="@+id/tengroup"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="them group"
+        />
+
+
+</RelativeLayout>
\ No newline at end of file
Index: app/src/main/java/com/example/tinnhn/Call/GroupHoiThoaiActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/example/tinnhn/Call/GroupHoiThoaiActivity.java	(date 1606123565309)
+++ app/src/main/java/com/example/tinnhn/Call/GroupHoiThoaiActivity.java	(date 1606123565309)
@@ -0,0 +1,57 @@
+package com.example.tinnhn.Call;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.widget.Toast;
+
+import com.example.tinnhn.R;
+import com.sinch.android.rtc.calling.Call;
+
+public class GroupHoiThoaiActivity extends BaseActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_group_hoi_thoai);
+    }
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // R.menu.mymenu is a reference to an xml file named mymenu.xml which should be inside your res/menu directory.
+        // If you don't have res/menu, just create a directory named "menu" inside res
+        getMenuInflater().inflate(R.menu.callgroup, menu);
+        return super.onCreateOptionsMenu(menu);
+    }
+
+    // handle button activities
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        int id = item.getItemId();
+
+        if (id == R.id.groupchat) {
+            //đang nhap sinchclient o day = UserEmail sau do goi cho EmailNguoiGui
+
+            if (getGiaodiendichvu().isStarted()) {
+
+                Hamchuyenidgroupquagroupcall();
+
+            } else {
+                Toast.makeText(this, "chua chay dich vu", Toast.LENGTH_SHORT).show();
+            }
+        }
+        return super.onOptionsItemSelected(item);
+    }
+    private void Hamchuyenidgroupquagroupcall() {
+        String username="idgroup";
+
+        Intent callScreen = new Intent(this, CuocGoi_Screen.class);
+        callScreen.putExtra(SinchServices.CALL_ID,username);
+        startActivity(callScreen);
+
+//        Intent mainActivity = new Intent(this, Dialer.class);
+//        startActivity(mainActivity);
+    }
+}
\ No newline at end of file
Index: app/src/main/res/drawable/ic_groupchat.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/drawable/ic_groupchat.xml	(date 1606122933584)
+++ app/src/main/res/drawable/ic_groupchat.xml	(date 1606122933584)
@@ -0,0 +1,5 @@
+<vector android:height="24dp" android:tint="#F2F2F2"
+    android:viewportHeight="24" android:viewportWidth="24"
+    android:width="24dp" xmlns:android="http://schemas.android.com/apk/res/android">
+    <path android:fillColor="@android:color/white" android:pathData="M12,1c-4.97,0 -9,4.03 -9,9v7c0,1.66 1.34,3 3,3h3v-8H5v-2c0,-3.87 3.13,-7 7,-7s7,3.13 7,7v2h-4v8h4v1h-7v2h6c1.66,0 3,-1.34 3,-3V10c0,-4.97 -4.03,-9 -9,-9z"/>
+</vector>
Index: app/src/main/res/menu/callgroup.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/menu/callgroup.xml	(date 1606122948144)
+++ app/src/main/res/menu/callgroup.xml	(date 1606122948144)
@@ -0,0 +1,10 @@
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+    <item
+        android:id="@+id/groupchat"
+        android:title=""
+        app:showAsAction="always"
+        android:icon="@drawable/ic_groupchat"
+        />
+
+</menu>
\ No newline at end of file
Index: app/src/main/res/layout/activity_group_hoi_thoai.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/activity_group_hoi_thoai.xml	(date 1606123087149)
+++ app/src/main/res/layout/activity_group_hoi_thoai.xml	(date 1606123087149)
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".HoiThoaiActivity"
+    android:background="#fff"
+    >
+
+    <ListView
+        android:id="@+id/list_Hoithoai"
+        android:layout_width="match_parent"
+        android:layout_height="538dp"
+        android:layout_above="@+id/edtNoiDung"
+        android:layout_alignParentStart="true"
+        android:layout_alignParentLeft="true"
+        android:layout_alignParentRight="true"
+        android:layout_alignParentTop="true"
+        android:layout_marginStart="0dp"
+        android:layout_marginLeft="0dp"
+        android:layout_marginTop="0dp"
+        android:divider="@null" />
+    <EditText
+        android:id="@+id/edtNoiDung"
+        android:layout_width="333dp"
+        android:layout_height="wrap_content"
+        android:layout_alignParentStart="true"
+        android:layout_alignParentLeft="true"
+        android:layout_alignParentBottom="true"
+        android:layout_marginStart="0dp"
+        android:layout_marginLeft="0dp"
+        android:layout_marginRight="30dp"
+        android:layout_marginBottom="0dp"
+        android:background="#fff"
+        android:hint="Nội dung"
+        android:maxLines="1"
+        android:lines="1"
+        android:padding="20dp" />
+
+    <Button
+        android:id="@+id/btbGui"
+        android:layout_width="75dp"
+        android:layout_height="45dp"
+        android:layout_alignParentEnd="true"
+        android:layout_alignParentRight="true"
+        android:layout_alignParentBottom="true"
+        android:layout_marginStart="80dp"
+        android:layout_marginLeft="80dp"
+        android:layout_marginEnd="3dp"
+        android:layout_marginRight="3dp"
+        android:layout_marginBottom="5dp"
+        android:background="@drawable/radius"
+        android:text="Send"
+        android:textColor="#fff" />
+
+
+</RelativeLayout>
\ No newline at end of file
Index: app/src/main/java/com/example/tinnhn/Call/GroupCallActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/example/tinnhn/Call/GroupCallActivity.java	(date 1606123870612)
+++ app/src/main/java/com/example/tinnhn/Call/GroupCallActivity.java	(date 1606123870612)
@@ -0,0 +1,16 @@
+package com.example.tinnhn.Call;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.os.Bundle;
+
+import com.example.tinnhn.R;
+
+public class GroupCallActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_group_call);
+    }
+}
\ No newline at end of file
Index: app/src/main/res/layout/activity_group_call.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/activity_group_call.xml	(date 1606123992533)
+++ app/src/main/res/layout/activity_group_call.xml	(date 1606123992533)
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".Call.GroupCallActivity">
+
+</androidx.constraintlayout.widget.ConstraintLayout>
\ No newline at end of file
